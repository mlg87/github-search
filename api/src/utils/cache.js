const memoryCache = require('memory-cache')

module.exports = duration => (req, res, next) => {
  let key = '__express__' + req.originalUrl || req.url
  let cachedBody = memoryCache.get(key)
  if (cachedBody) {
    res.send(cachedBody)
    return
  } else {
    res.sendResponse = res.send
    res.send = body => {
      memoryCache.put(key, body, duration * 1000) // seconds
      res.sendResponse(body)
    }
    next()
  }
}
