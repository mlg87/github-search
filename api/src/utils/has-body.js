module.exports = (req, res, next) => {
  // TODO improve the check
  if (!Object.keys(req.body).length) {
    res.status(400).send('incorrect body on request')
  } else {
    next()
  }
}
