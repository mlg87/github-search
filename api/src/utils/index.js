const cache = require('./cache')
const hasBody = require('./has-body')
const hasParam = require('./has-param')
const hasQuery = require('./has-query')

module.exports = {
  cache,
  hasBody,
  hasParam,
  hasQuery
}
