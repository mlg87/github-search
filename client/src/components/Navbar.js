import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'

import '../styles/Navbar.css'

class Navbar extends Component {
  renderLinks() {
    const routes = [
      { path: '/', label: 'Home' },
      { path: '/search', label: 'Search' },
      { path: '/about', label: 'About' }
    ]
    return routes.map(({ path, label }) => {
      return (
        <NavLink to={path} activeClassName="app-nav-link--selected" className="app-nav-link">
          {label}
        </NavLink>
      )
    })
  }

  render() {
    return (
      <nav className="app-nav">
        <ul>{this.renderLinks()}</ul>
      </nav>
    )
  }
}

export default Navbar
