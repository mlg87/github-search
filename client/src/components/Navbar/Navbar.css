.nav {
  height: 80px;
  background: linear-gradient(to right, #f09a38, #d888e2);
  display: flex;
  flex-grow: 1;
  justify-content: center;
  align-items: center;
}

.nav__link {
  margin-right: 20px;
  margin-left: 20px;
  text-decoration: none;
  color: #fff;
  font-size: 28px;
  font-weight: 600;
}

.nav__link--selected {
  border-bottom: 2px solid #fff;
}
