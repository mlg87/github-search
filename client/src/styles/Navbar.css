.app-nav {
  height: 80px;
  background: linear-gradient(to right, #f0a202, #ffd289);
  display: flex;
  flex-grow: 1;
  justify-content: center;
  align-items: center;
}

.app-nav-link {
  padding-right: 20px;
  padding-left: 20px;
  text-decoration: none;
  color: '#fff';
}

.app-nav-link--selected {
  color: #aaa;
}
