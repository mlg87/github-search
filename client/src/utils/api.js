const API_URL = process.env.REACT_APP_API_URL

export default class API {
  async fetch(url, method = 'GET', body = null) {
    try {
      return await fetch(`${API_URL}/${url}`, {
        method,
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        },
        body: body ? JSON.stringify(body) : body
      })
    } catch (error) {
      throw Error(error)
    }
  }
}
