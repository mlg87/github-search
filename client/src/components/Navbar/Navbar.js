import React, { Component } from 'react' // eslint-disable-line
import { NavLink } from 'react-router-dom' // eslint-disable-line

import './Navbar.css'

class Navbar extends Component {
  renderLinks() {
    const routes = [
      { path: '/', label: 'Search' },
      { path: '/results', label: 'Results' },
      { path: '/about', label: 'About' }
    ]
    return routes.map(({ path, label }) => {
      return (
        <NavLink
          to={path}
          activeClassName="nav__link--selected"
          className="nav__link"
          exact
          key={`${path}`}
        >
          {label}
        </NavLink>
      )
    })
  }

  render() {
    return (
      <nav className="nav">
        <ul>{this.renderLinks()}</ul>
      </nav>
    )
  }
}

export default Navbar
