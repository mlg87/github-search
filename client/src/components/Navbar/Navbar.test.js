import React from 'react'
import { BrowserRouter, NavLink } from 'react-router-dom'
import Navbar from './Navbar'
import { mountWrap, shallowWrap } from '../../utils/testContextWrap'
import renderer from 'react-test-renderer'

describe('Navbar', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders three <NavLink />s', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    )

    expect(
      wrapper
        .dive()
        .dive()
        .find(NavLink).length
    ).toBe(3)
  })
})
