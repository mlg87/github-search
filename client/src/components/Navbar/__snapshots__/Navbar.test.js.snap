// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <NavLink
                activeClassName="nav__link--selected"
                ariaCurrent="true"
                className="nav__link"
                exact={true}
                to="/"
        >
                Search
        </NavLink>
        <NavLink
                activeClassName="nav__link--selected"
                ariaCurrent="true"
                className="nav__link"
                exact={true}
                to="/results"
        >
                Results
        </NavLink>
        <NavLink
                activeClassName="nav__link--selected"
                ariaCurrent="true"
                className="nav__link"
                exact={true}
                to="/about"
        >
                About
        </NavLink>
</ul>,
      "className": "nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="nav__link--selected"
            ariaCurrent="true"
            className="nav__link"
            exact={true}
            to="/"
>
            Search
</NavLink>,
          <NavLink
            activeClassName="nav__link--selected"
            ariaCurrent="true"
            className="nav__link"
            exact={true}
            to="/results"
>
            Results
</NavLink>,
          <NavLink
            activeClassName="nav__link--selected"
            ariaCurrent="true"
            className="nav__link"
            exact={true}
            to="/about"
>
            About
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "nav__link--selected",
            "ariaCurrent": "true",
            "children": "Search",
            "className": "nav__link",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/results",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "nav__link--selected",
            "ariaCurrent": "true",
            "children": "Results",
            "className": "nav__link",
            "exact": true,
            "to": "/results",
          },
          "ref": null,
          "rendered": "Results",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/about",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "nav__link--selected",
            "ariaCurrent": "true",
            "children": "About",
            "className": "nav__link",
            "exact": true,
            "to": "/about",
          },
          "ref": null,
          "rendered": "About",
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <NavLink
                    activeClassName="nav__link--selected"
                    ariaCurrent="true"
                    className="nav__link"
                    exact={true}
                    to="/"
          >
                    Search
          </NavLink>
          <NavLink
                    activeClassName="nav__link--selected"
                    ariaCurrent="true"
                    className="nav__link"
                    exact={true}
                    to="/results"
          >
                    Results
          </NavLink>
          <NavLink
                    activeClassName="nav__link--selected"
                    ariaCurrent="true"
                    className="nav__link"
                    exact={true}
                    to="/about"
          >
                    About
          </NavLink>
</ul>,
        "className": "nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="nav__link--selected"
              ariaCurrent="true"
              className="nav__link"
              exact={true}
              to="/"
>
              Search
</NavLink>,
            <NavLink
              activeClassName="nav__link--selected"
              ariaCurrent="true"
              className="nav__link"
              exact={true}
              to="/results"
>
              Results
</NavLink>,
            <NavLink
              activeClassName="nav__link--selected"
              ariaCurrent="true"
              className="nav__link"
              exact={true}
              to="/about"
>
              About
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "/",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav__link--selected",
              "ariaCurrent": "true",
              "children": "Search",
              "className": "nav__link",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/results",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav__link--selected",
              "ariaCurrent": "true",
              "children": "Results",
              "className": "nav__link",
              "exact": true,
              "to": "/results",
            },
            "ref": null,
            "rendered": "Results",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/about",
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav__link--selected",
              "ariaCurrent": "true",
              "children": "About",
              "className": "nav__link",
              "exact": true,
              "to": "/about",
            },
            "ref": null,
            "rendered": "About",
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
